libs/remix-ui/solidity-compiler/src/lib/solidity-compiler.tsx

```diff
@@ -40,6 +42,14 @@ export const SolidityCompiler = (props: SolidityCompilerProps) => {
     })()
   }, [])
 
+  useEffect(() => {
+    if (badgeStatus[currentFile]) {
+      api.emit('statusChanged', badgeStatus[currentFile])
```

Comment:
can't you use the `statusChanged` function too here

Reply:
`statusChanged` function wouldn't work here. A plugin emitted event is needed.